Assets/package/Runtime/Prefabs/StandardPrefabClasses/PointListClusteredParticlesPrefab.cs

```diff
@@ -0,0 +1,172 @@
+﻿/*
+ * Copyright (c) 2019 Robotic Eyes GmbH software
+ *
+ * THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY
+ * KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
+ * PARTICULAR PURPOSE.
+ *
+ */
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using UnityEngine;
+using Random = UnityEngine.Random;
+using ColoredPoint = RoboticEyes.Rex.RexFileReader.Examples.ThreadedClustering.ColoredPoint;
+
+namespace RoboticEyes.Rex.RexFileReader.Examples
+{
+	public class PointListClusteredParticlesPrefab : RexPointListObject
+	{
+		[SerializeField] 
+		private bool dynamicCloudDensityEnabled;
+		[SerializeField] 
+		private int maxAllowedParticlesOnScreen = (int)10e5;
+		[SerializeField] 
+		private int averageParticlesPerCluster = (int)60e3;
+		[SerializeField] 
+		private GameObject pointListParticlesPrefab;
+		private int clusterCount;
+		private float baseReductionFactor;
+		private float minParticleSize;
+		private PointListParticlesPrefab[] childPointListParticles;
+
+		public override bool SetPoints (List<Vector3> pointPositions, List<Color> pointColors)
+		{
+			HandleStartup (ref pointPositions);
+            
+            var startingCentroids = GetRandomStartingCentroids (pointPositions, clusterCount);
+			var coloredPoints = CombinePointInfo (pointPositions, pointColors);
+
+			StartClusteringCoroutine (coloredPoints, startingCentroids, (success, clusteredPointPositions) =>
+				{
+					if (!success) return;
+					StartCoroutine (ParticleSystemSpawnerCoroutine (clusteredPointPositions));
+				}
+			);
+
+			return true;
+		}
+        private void HandleStartup (ref List<Vector3> pointPositions)
+        {
+            baseReductionFactor = Mathf.Min (1, (float)maxAllowedParticlesOnScreen / pointPositions.Count);
+            clusterCount = Mathf.CeilToInt ((float)pointPositions.Count / averageParticlesPerCluster);
+
+            if (maxAllowedParticlesOnScreen >= pointPositions.Count) dynamicCloudDensityEnabled = false;
+
+            if (pointPositions.Count > maxAllowedParticlesOnScreen) {
+                print ($"point cloud of {pointPositions.Count} reduced to {maxAllowedParticlesOnScreen} particles");
+            }
+        }
+
+        private IEnumerator ParticleSystemSpawnerCoroutine (IEnumerable<List<ColoredPoint>> clusteredPointPositions)
+		{
+			foreach (var clusteredPointPosition in clusteredPointPositions)
+			{
+				var child = Instantiate (pointListParticlesPrefab, transform).GetComponent<PointListParticlesPrefab> ();
+				
+				child.SetPoints (clusteredPointPosition);
+				child.SetDensity (baseReductionFactor);
+
+				yield return null;
+			}
+
+			childPointListParticles = GetComponentsInChildren<PointListParticlesPrefab> ();
```

Comment:
Be careful, this would only return active Children. If you also want inactive one use `GetComponentsInChildren(true)`